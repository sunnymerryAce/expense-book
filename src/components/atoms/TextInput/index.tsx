import React from "react";
import TextField from "@material-ui/core/TextField";
import useStyles from "./style";
export interface TextInputProps {
  label: string;
  type?: string;
  error: boolean;
  defaultValue?: string;
  helperText?: string;
  handleChange: (...props: any[]) => any;
  className?: string;
  disabled?: boolean;
}

const TextInput: React.FC<TextInputProps> = (props) => {
  const classes = useStyles();

  return (
    <TextField
      variant="outlined"
      onChange={(e): void => {
        props.handleChange(e.target.value);
      }}
      label={props.label}
      defaultValue={props.defaultValue || ""}
      error={props.error}
      helperText={props.helperText}
      type={props.type || "text"}
      className={`TextInput ${props.className && props.className} ${
        classes.root
      }`}
      disabled={props.disabled || false}
    />
  );
};

export default TextInput;
