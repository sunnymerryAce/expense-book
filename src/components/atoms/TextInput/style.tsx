import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      width: "calc(50% - 8px)",
      marginTop: "20px",
      "& > *": {
        margin: `${theme.spacing(1)} 0`,
      },
    },
  })
);

export default useStyles;
