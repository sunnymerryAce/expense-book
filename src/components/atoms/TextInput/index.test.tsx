import React from "react";
import TextInput from "./";
import { shallow, mount } from "enzyme";

describe("<TextInput />", () => {
  let container: any = null;
  const props = {
    label: "Budget",
    defaultValue: "25000",
    error: false,
    type: "number",
    handleChange: jest.fn(),
  };

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    container = shallow(<TextInput {...props} />);
    expect(container.html()).toMatchSnapshot();
  });

  test("should have proper label", () => {
    container = mount(<TextInput {...props} />);
    expect(container.find("label.MuiInputLabel-root").text()).toBe(props.label);
  });

  test("should have proper default value", () => {
    container = mount(<TextInput {...props} />);
    expect(
      container.find("input.MuiInputBase-input").get(0).props.defaultValue
    ).toBe(props.defaultValue);
  });

  test("should be proper input type", () => {
    container = mount(<TextInput {...props} />);
    expect(container.find("input.MuiInputBase-input").get(0).props.type).toBe(
      props.type
    );
  });

  test("should be error", () => {
    const props = {
      label: "Budget",
      defaultValue: "25000",
      error: true,
      helperText: "helperText",
      handleChange: jest.fn(),
    };
    container = mount(<TextInput {...props} />);
    expect(container.exists(".MuiInputBase-root.Mui-error")).toBe(true);
  });

  test("should have proper error text", () => {
    const props = {
      label: "Budget",
      defaultValue: "25000",
      error: true,
      helperText: "helperText",
      handleChange: jest.fn(),
    };
    container = mount(<TextInput {...props} />);
    expect(container.find(".MuiFormHelperText-root").text()).toBe(
      props.helperText
    );
  });

  test("event handler should be triggered", () => {
    container = shallow(<TextInput {...props} />);
    container
      .find(".TextInput")
      .simulate("change", { target: { value: "test" } });
    expect(props.handleChange).toBeCalled();
  });
});
