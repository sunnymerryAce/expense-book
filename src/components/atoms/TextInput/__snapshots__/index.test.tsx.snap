// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match the snapshot 1`] = `"<div class=\\"MuiFormControl-root MuiTextField-root TextInput undefined makeStyles-root-1\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled\\" data-shrink=\\"true\\">Budget</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input type=\\"number\\" aria-invalid=\\"false\\" value=\\"25000\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-60 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-62 PrivateNotchedOutline-legendNotched-63\\"><span>Budget</span></legend></fieldset></div></div>"`;
